### PR TITLE
Dynfailover

### PR DESCRIPTION
**Pull request by davidoae**
Suggestions please, keep an eye out for edge conditions I've missed :)
I set this for the monitor server but now I'm wondering if we should have it on puppet server instead.

### CODE REVIEW

File: environments/production/modules/localconfig/manifests/dynfailover.pp [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    $dynapiuser  = hiera('dynapiuser')
+    $dynapipass  = hiera('dynapipass')
+    $web0ip = hiera('web0ip')
+    $web1ip = hiera('web1ip')
+
+    file ( 'dyn_failover':
+        path    => "/usr/local/bin/dyn_failover",
+        mode    => 0755,
+        content => template('localconfig/dyn_failover.erb')
+    }
```
Comment (mrvisser) [inline environments/production/modules/localconfig/manifests/dynfailover.pp]: does this work? There is an opening `(` and a closing `}`

**Comment by mrvisser** [comment]
The script looks good to me, though I don't think this is ready for merge yet. There are a few outstanding items:

1. Did the bracket mismatch result in a compilation failure of the templates? If so, then there is obviously some more testing needed.
2. The configuration values needed by the puppet scripts should be added to the JSON heira configs, while the username / password would need to be placed in the hiera_secure variant -- I believe this is needed for the puppet scripts to work as well.
3. We should have a nagios alert to let us know the existence of the lock file so we can remember to re-enable the failover polling
4. We should have a nagios alert to let us know if the state file is set to 'bad' so we can be warned if a web server switch is imminent or flapping for some reason. Preferably, the nagios polling frequency can be lower than our failover timeout frequency so we do have warning.

**Comment by mrvisser** [comment]
Assigning back to @davidoae for feedback or more changes

**Comment by davidoae** [comment]
Note that when deployed it's not actually turned on until DNS is changed so there's flexibility to test in production.
1. I'm sure it would, I believe it was a duplication error, another test will be done before PR is re-submitted.
2. This is already added to common_hiera_secure.json from previous test.
3. Seems reasonable, will do.
4. This depends how long we accept failures before performing the switch. I might be able to have the script itself send updates to nagios so we're not trying to squeeze checks in a window only a few mins long.

**Comment by davidoae** [comment]
I have had some issues with item 4, I think we should still deploy and maybe add this later.
Also note that once merged it will not be actually activated until DNS is manually updated to use it.

**Pull request merged**